### ISSUE TITLE
Enable autoplay for [S] Act 7 and [S] Collide and hide controls

### ISSUE THREAD

**Issue by Bodertz** [Bambosh/unofficial-homestuck-collection#424]
The Collide and Act 7 animations, unlike most animations in Homestuck, are videos rather than flash animations.  As it stands now, this means they are presented unlike any other major animation in Homestuck.  I think this is unfortunate.  Subjectively, it feels cheep.  It feels like it's not fully integrated into the page.  I understand that this is how it was originally released (as a YouTube embed, no less), but I think the presentation could be improved in the UHSC by making it behave more like a regular flash animation.  Doing that means two things:

1. Disabling video controls (since no other animation has them)
2. Enabling autoplay (since most animations autoplay)

I've been able to make this happen by editing this line to include `autoplay` and remove `controls` and `controlsList="nodownload"`:

https://github.com/Bambosh/unofficial-homestuck-collection/blob/ef9cd58c6ad8221ab3df040dec32c1204deadcf0/src/components/UIElements/MediaEmbed.vue#L4

This works except it has the unintended side effect of disabling video controls in places it shouldn't such as the link to a clip of Con Air on `/mspa/001920` (John: Examine Con Air poster).  I don't imagine this is an unsolvable problem, however.  Is there some way to check the name of the file and alter the behaviour based on that?

Of course, I'm sure some people like the video controls.  They are long animations after all.  Maybe other people don't feel that the presentation is as harmed by the presence of a progress bar as I do, and they'd hate for the controls to be removed.  An option in the settings would be ideal, so each person can make their own choice here.  Maybe I'd be the only one who would switch the controls off, but I don't think so.  Unless you have Flash or Ruffle or some other thing like that installed, the official website has YouTube embeds even for things like `Terezi: Reme8er` or `Game Over`, and I think anyone could disagree that that's worse.  I think it's a similar principle for Collide and Act 7, just not as severe.

Would anyone object to this being an option?

**Comment by Bodertz** [comment]
I see that at least the autoplay part is mentioned in this pull request: https://github.com/Bambosh/unofficial-homestuck-collection/pull/262/

**Comment by GiovanH** [comment]
I would definitely recommend looking at #262 as a step towards making Collide a bit nicer. I hadn't considered doing something like that with Act 7 (since the format is 100% video animation, and not flash-like), but it's possible.

Both of these pages were originally presented as videos, and it's hard to deny the utility of having the player controls and seek functionality, so I think it's unlikely we'll remove that entirely (although perhaps a mod could).

**Comment by Bodertz** [comment]
I've built that version and I would say it's an improvement.  Pausing it at the right moment is neat.  I still dislike the progress bar and would prefer it not be there, but I'm sure that doesn't surprise you.

I of course agree that the progress bar is useful, but I don't think the usefulness is worth it.  Cascade is of a similar length to Collide (14 minutes vs 18 minutes), and I don't think that extra 4 minutes is enough to make the progress bar especially more useful for Collide.  I think it's more an accident of history that one was released as a video and one as a flash animation.  I don't think it's an artistic statement or anything like that.

I can understand that it may be too small a change to make a settings page for.  Perhaps the collection could include the YouTube versions of each animation (end of Act 1, end of Act 2, Jack: Ascend, and so on) and there could be a toggle in the settings for those who want progress bars in those animations as well.  That would make the option more sensible to include as it would not just be Collide and Act 7 that are affected.  But I don't know if anyone would actually want that.

**Comment by Bambosh** [comment]
It's worth noting that while Cascade didn't have a progress bar, it did have a pause button and a chapter select - something Collide would have also had if it were an actual Flash animation. This thing's almost 20 minutes long, and at that point it becomes detrimental to hold back player functionality to keep the bit going.

**Comment by Bodertz** [comment]
I agree a pause button would be useful.  I think a chapter selection is also doable.

We disagree on the value that upholding the bit adds.  The comic, especially by the end, made an effort to be integrated into the website in surprising ways.  Flash animations would start and then expand past what seemed to be the border of the video as in Cascade or Game Over or Remem8er or A6A6I5 (She's 8ack).  Other animations, of course, did not do this.  But it felt like they could, once the comic told us it's a thing that could happen.  Sure, there are clues if you look out for them.  The font in the animation might not match what you have on your system, for example (I've installed Courier New, Verdana, and Comic Sans in Linux to try to get around this as much as possible -- I assume there are legal issues with embedding them in the Unofficial Homestuck Collection?).  And you can't select text in the animation, so you'll know whether or not something will happen with the title by whether or not you can select it.  But that's easy to miss in a way that an embedded video just isn't.  An embedded video with a visible progress bar tells me the bounds of the animation before it's started.  It feels like a video plopped onto the page rather than the format defying experience of some of the latter half of the comic.  Sure, there are benefits to having a progress bar.  Twenty minutes is a long time.  But it's not that long, really.  It's not a movie or a play, and people have suffered through those without the ability to seek or even pause.  I think upholding the bit as you describe it has more benefits than downsides.

But like I said, we disagree on that.  What we might agree on is autoplaying Act 7.  Do you oppose that as well?
